authorizations/message-filter/src/contract.rs

```diff
@@ -0,0 +1,408 @@
+use cosmwasm_std::{entry_point, to_binary, Addr, Uint128};
+use cosmwasm_std::{Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+use cw_auth_middleware::msg::IsAuthorizedResponse;
+use schemars::{JsonSchema, Map};
+use serde_derive::{Deserialize, Serialize};
+use serde_json_wasm::{from_str, to_string};
+
+use crate::error::ContractError;
+use crate::msg::{AuthorizationsResponse, ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{Authorization, Config, Kind, ALLOWED, CONFIG};
+use cw_auth_middleware::ContractError as AuthorizationError;
+
+const CONTRACT_NAME: &str = "crates.io:whitelist";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
+pub enum Value {
+    Null,
+    Bool(bool),
+    U128(Uint128),
+    Number(u64), // Why doesn't u128 work here?
+    String(String),
+    Array(Vec<Value>),
+    Object(Map<String, Value>),
```

Comment:
I am quite worried about using Map in any serde serialisable struct.
Also, very odd to see this come from schemars.

This *may* be safe, but needs some very careful checking.
Minimally, check the json encoding "manually" in test cases.
Also ensure that when this compiles to wasm, it passes `check_contract`
([see this example](https://github.com/CosmWasm/cw-plus/blob/main/.circleci/config.yml#L602-L618))

Reply:
Ran the checks and they seem to pass. I can run them on the optimized contract later.
<img width="434" alt="__devel_dao-contracts" src="https://user-images.githubusercontent.com/37868/182824755-e2c7d1ca-9a07-4523-bcc9-c1770b0fb6d7.png">

I've tested this manually and it works on the chain, and schemars uses https://docs.rs/indexmap/latest/indexmap/ internally to preserve ordering. I can add some more tests to the serialization/deserialization specifically focusing on ordering (is non-determinism your main concern?)